### PR TITLE
feat(MCSPAuthenticator): add new authenticator for Multi-Cloud Saas Platform

### PR DESCRIPTION
**Pull request by padamstx**
This commit introduces the new MCSPAuthenticator that can be used to exchange an apikey for an MCSP access token using the Multi-Cloud Saas Platform authentication token server's 'POST /siusermgr/api/1.0/apikeys/token' operation.

### CODE REVIEW

File: src/test/java/com/ibm/cloud/sdk/core/test/security/Cp4dAuthenticatorTest.java [hunk 1]
```diff
@@ -339,7 +339,7 @@ public void testAuthenticateNewAndStoredToken() {
     assertEquals(authenticator.getClient(), client);
 
     // Authenticator should request new, valid token.
-     Request.Builder requestBuilder = new Request.Builder().url("https://test.com");
```
Comment (padamstx) [inline src/test/java/com/ibm/cloud/sdk/core/test/security/Cp4dAuthenticatorTest.java:342]: This was for you, @pyrooka :joy:

**Review by pyrooka — APPROVED** [review]
Look pretty good - as far as I can tell!

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

**Pull request merged**